pages/_app.tsx

```diff
@@ -42,7 +43,12 @@ const App = ({ Component, pageProps }) => {
   return (
     <>
       <DefaultSeo {...NEXT_SEO_DEFAULT} />
-      <Component {...pageProps} />
+      {!isDev && (
+        <AzureAppInsights>
+          <Component {...pageProps} />
+        </AzureAppInsights>
+      )}
+      {isDev && <Component {...pageProps} />}
```

Comment:
I would've expected that the container would always be setup and app insights enabled/disabled internally as required

Reply:
done - so instead of checking it here we are loading the appInsight based on the env variable